pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -327,3 +334,79 @@ func getKubernetesResources(kogitoRes *kogitores.KogitoAppResources) []utilsres.
 
 	return k8sRes
 }
+
+// Ensure that all Kogito images are available before the build.
+func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoApp) (reconcile.Result, error) {
+	isISPresentOnOpenshiftNamespace := true
+
+	imageStreamTag := kogitores.ImageStreamTag
+	if instance.Spec.Build.ImageS2I.ImageStreamTag != "" {
+		imageStreamTag = instance.Spec.Build.ImageS2I.ImageStreamTag
+		log.Infof("Using ImageS2I Tag from CR: %s", imageStreamTag)
+	}
+
+	// First look on the current namespace, if the imagestreams are available, return
+	// and don't check for anything else.
+	for _, key := range kogitores.ImageStreamNameList {
+		is, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: instance.Namespace}, imageStreamTag)
+		if is == nil {
+			r.isImagestreamReady = false
+		}
+	}
+
+	// look for the Kogito image stream on the default namespace (openshift)
+	// All images must be present, if not create it on current namespace
+	if !r.isImagestreamReady {
+		for _, key := range kogitores.ImageStreamNameList {
+			isOcpNs, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: kogitores.ImageStreamNamespace}, imageStreamTag)
+			if isOcpNs == nil {
+				isISPresentOnOpenshiftNamespace = false
+			}
+		}
+	}
+
+	if !isISPresentOnOpenshiftNamespace && !r.isImagestreamReady {
+
+		for _, is := range kogitores.KogitoImageStream(instance.Namespace, imageStreamTag).Items {
+			is.Namespace = instance.Namespace
+			hasIs, err := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: is.Name, Namespace: instance.Namespace}, imageStreamTag)
+			if err != nil {
+				log.Error(err.Error())
+			}
+
+			if hasIs == nil {
+				r.isImagestreamReady = false
+				_, err := openshift.ImageStreamC(r.client).CreateImageStream(&is)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+
+				// wait for 5 seconds to the image get ready
+				tagRef := fmt.Sprintf("%s:%s", is.Name, imageStreamTag)
+				counter := 0
+				for {
+					imageTag, getISerr := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: is.Name, Namespace: is.Namespace}, imageStreamTag)
+					if getISerr != nil {
+						log.Error(getISerr)
+					}
+					if imageTag != nil && imageTag.Image.Name != "" {
+						log.Infof("Image %s ready.", imageTag.Name)
+						break
+					}
+					if counter >= 10 {
+						return reconcile.Result{}, fmt.Errorf("Failed to get information about image tag ref %s on namespace %s", tagRef, is.Namespace)
+					}
+					counter++
+					time.Sleep(500 * time.Millisecond)
```

Comment:
Maybe `time.Sleep(5 * time.Second)`?

Reply:
if we change to 5*seconds, the for loop will wait for 50 seconds in total if no IS is created and 5 seconds for every try.
with 0.5s it will wait 5 seconds tops before thrown an error and 0.5 seconds between every try.

Reply:
Oh ok, you're right I misread the parameter.